plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2932,17 +2936,29 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
                 .orElseThrow(() -> new IllegalStateException("Storage table missing in definition of materialized view " + materializedViewName));
 
         Table icebergTable = catalog.loadTable(session, storageTableName);
-        String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
+        Snapshot currentSnapshot = icebergTable.currentSnapshot();
+        String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnTableFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                .orElse("");
+
         if (dependsOnTables.isEmpty()) {
+            // If the source of the MV is a PTF then source tables will be empty but source functions won't
+            if (!dependsOnTableFunctions.isEmpty()) {
+                Instant refreshTime = Optional.ofNullable(icebergTable.currentSnapshot().summary().get(TRINO_QUERY_START_TIME))
+                        .map(Instant::parse)
+                        .orElseGet(() -> Instant.ofEpochMilli(icebergTable.currentSnapshot().timestampMillis()));
+                return new MaterializedViewFreshness(UNKNOWN, Optional.of(refreshTime));
+            }
             // Information missing. While it's "unknown" whether storage is stale, we return "stale".
-            // Normally dependsOnTables may be missing only when there was no refresh yet.
+            // Normally dependsOnTables may be missing only when there was no refresh yet,  when the source is a PTF or when there is no source for example: CREATE M V foo AS VALUES 1
```

Comment:
"dependsOnTables may be missing ...." and then we talk about case when dependsOnTables is not missing, but it's empty.

Reply:
Yes but with current implementation it will always be missing before first REFRESH. Then it will always contain a string(can be empty)

Reply:
so my understanding is :
1. when source is table but there was no refresh so this dependsOnTables is missing, then the view is stale - so we are fine
2. when source is ptf dependsOnTables is missing and we should never get here - so we are fine
3. when source is constant like in `CREATE M V test AS VALUES 1` then dependsOnTables will be empty

Only point 3 is interesting now. But I don't know what should be the result. If the view has not be refreshed then it is stale so this part is fine. I am not sure what should happen in opposite case ?